services/director-v2/src/simcore_service_director_v2/utils/dags.py

```diff
@@ -0,0 +1,46 @@
+import logging
+from typing import List
+
+import networkx as nx
+from models_library.project_nodes import NodeID
+from models_library.projects import Workbench
+
+from .computations import to_node_class, NodeClass
+
+log = logging.getLogger(__file__)
+
+
+def find_entrypoints(graph: nx.DiGraph) -> List[NodeID]:
+    entrypoints = [n for n in graph.nodes if not list(graph.predecessors(n))]
+    log.debug("the entrypoints of the graph are %s", entrypoints)
+    return entrypoints
+
+
+def create_dag_graph(workbench: Workbench) -> nx.DiGraph:
+    dag_graph = nx.DiGraph()
+    for node_id, node in workbench.items():
+        if to_node_class(node.key) == NodeClass.COMPUTATIONAL:
```

Comment:
We are still treating the entire workbench as an individual graph? If there is a mix of computational and dynamic services, where the computation ones are split by a dynamic service, a single graph is formed and all the comp tasks run as a unique graph?

Reply:
well you can see that this is only partially the case.

if your 2 graphs are separated by a dynamic service (which btw is not officially supported yet). 2 DAGs will be generated without dependencies. so what will happen is that both will run in parallel, which will be a failure.
This PR does not fix everything yet.

Reply:
OK, good to know.